alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractJobResourceActions.java

```diff
@@ -0,0 +1,211 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.action.api;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import org.springframework.http.HttpStatus;
+
+import com.synopsys.integration.alert.common.action.ActionResponse;
+import com.synopsys.integration.alert.common.action.ValidationActionResponse;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.enumeration.DescriptorType;
+import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.persistence.accessor.DescriptorAccessor;
+import com.synopsys.integration.alert.common.persistence.model.RegisteredDescriptorModel;
+import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.rest.model.JobFieldModel;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+
+public abstract class AbstractJobResourceActions implements JobResourceActions, ValidateAction<JobFieldModel>, TestAction<JobFieldModel> {
+    private static final EnumSet<DescriptorType> ALLOWED_JOB_DESCRIPTOR_TYPES = EnumSet.of(DescriptorType.PROVIDER, DescriptorType.CHANNEL);
+    private AuthorizationManager authorizationManager;
+    private DescriptorAccessor descriptorAccessor;
+
+    public AbstractJobResourceActions(AuthorizationManager authorizationManager, DescriptorAccessor descriptorAccessor) {
+        this.authorizationManager = authorizationManager;
+        this.descriptorAccessor = descriptorAccessor;
+        this.descriptorAccessor = descriptorAccessor;
+    }
+
+    protected abstract Optional<JobFieldModel> findJobFieldModel(UUID id);
+
+    protected abstract ActionResponse<JobFieldModel> createResource(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> deleteResource(UUID id);
+
+    protected abstract ActionResponse<List<JobFieldModel>> readAllResources();
+
+    protected abstract ValidationActionResponse testResource(JobFieldModel resource);
+
+    protected abstract ActionResponse<JobFieldModel> updateResource(UUID id, JobFieldModel resource);
+
+    protected abstract ValidationActionResponse validateResource(JobFieldModel resource);
```

Comment:
Same thought as the previous class. We should make it more explicit what the difference between something like `ActionResponse<JobFieldModel> create(JobFieldModel resource)` and `ActionResponse<JobFieldModel> createResource(JobFieldModel resource)` is.